homeassistant/components/proxmoxve/__init__.py

```diff
@@ -200,56 +202,65 @@ async def async_update_data() -> dict[str, Any] | None:
 
         def poll_api() -> dict[str, Any] | None:
             """Call the api."""
-            vm_status = call_api_container_vm(proxmox, node_name, vm_id, vm_type)
-            return vm_status
+            api_status = {}
 
-        vm_status = await hass.async_add_executor_job(poll_api)
-
-        if vm_status is None:
-            _LOGGER.warning(
+            try:
+                if api_category == ProxmoxType.QEMU:
+                    api_status = (
+                        proxmox.nodes(node_name).qemu(vm_id).status.current.get()
+                    )
+                elif api_category == ProxmoxType.LXC:
+                    api_status = (
+                        proxmox.nodes(node_name).lxc(vm_id).status.current.get()
+                    )
+            except AuthenticationError as error:
+                raise ConfigEntryAuthFailed from error
+            except SSLError as error:
+                raise UpdateFailed from error
+            except ConnectTimeout as error:
+                raise UpdateFailed from error
+            except ResourceException as error:
+                raise UpdateFailed from error
```

Comment:
Please combine known errors
```suggestion
            except (AuthenticationError, SSLError, ConnectTimeout, ResourceException) as error:
                raise UpdateFailed from error
```